src/metabase/query_processor/util/nest_query.clj

```diff
@@ -124,6 +124,20 @@
                 ::add/source-table  ::add/source
                 ::add/source-alias  source-alias))]))
 
+(defn- coerced-field?
+  [field-id]
+  (contains? (lib.metadata/field (qp.store/metadata-provider) field-id) :coercion-strategy))
+
+(defn- coercible-field-ref?
+  [form]
+  (and (vector? form)
+       (let [[tag id-or-name opts] form]
+         (and (= tag :field)
+              (not (:qp/ignore-coercion opts))
+              (or (contains? opts :temporal-unit)
+                  (and (int? id-or-name)
+                       (coerced-field? id-or-name)))))))
```

Comment:
This might be easier to read as a `lib.match/match-one`?

Reply:
I did it this way to make sure the field lookup comes last.